step-06-secure-secrets-in-key-vault/README.md

```diff
@@ -0,0 +1,90 @@
+# 05 - Setup GitHub Actions
+
+__This guide is part of the [migrate Java EE app to Azure training](../README.md)__
+
+How to store secrets in Azure Key Vault and access those secrets from your application. 
+
+## Overview of Key Vault References
+
+Azure Key Vault provides a centralized secret store with role-based access controls and audit history. Using App Service's **Key Vault References** feature, you can securely store your credentials, API keys, and other secrets in Key Vault and easily access them from your application code on App Service.
+
+[Key Vault References](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) is a feature of Azure App Service and Functions that allows your application code to retrieve secrets from Azure Key Vault without any code changes. The feature works by creating a system-assigned identity to authenticate to Key Vault, retrieve the secrets, and expose them as environment variables to your application on App Service. This means you can leverage all of Key Vault's secret management features without rewriting your application.
+
+## Create a Managed Identity
+
+A managed identity acts as a user in your Active Directory for automation purposes. It is inherently tied to your web app and will be deleted if the web app is deleted. For this scenario, the identity will be used to retrieve the secrets from Key Vault when the app starts. Run the following command to create a manged identity.
+
+```bash
+az webapp identity assign --name <your-jboss-webapp-name> --resource-group <rg-of-the-webapp>
+```
+
+The command wil output information about the identity. Copy the `principalId` for the next section.
+
+## Create a Key Vault
+
+Next, provision an Azure Key Vault, provide access to the identity you just created, and set the database username, password, and URL in Key Vault.
+
+1. Create the Key Vault using the Azure CLI.
+
+    ```bash
+    az keyvault create --name jboss-app-key-vault              \
+                        --resource-group <your_resource_group> \
+                        --location <location>                  \
+                        --enabled-for-deployment true          \
+                        --enabled-for-disk-encryption true     \
+                        --enabled-for-template-deployment true \
+                        --sku standard
+    ```
+
+1. Now grant the managed identity `get` and `list` access to the Key Vault.
+
+    ```bash
+    az keyvault set-policy --name java-app-key-vault     \
+                            --secret-permission get list  \
+                            --object-id <the principal ID from earlier>
+    ```
+
+1. Lastly, add the Postgres username, password, and URL to the Key Vault. If you followed sections 2 and 3, you should still have the secrets saved as environment variables on your machine.
+
+    ```bash
+    az keyvault secret set --name POSTGRES_SERVER_ADMIN_FULL_NAME \

```

Comment:
Vault name should be retrieved from an environment variable

Reply:
We should have 3 sub sections for each of the databases - SQL, PostgreSQL and MySQL